### ISSUE TITLE
Move `GmsegChallenge` outside of `sct-testing-large`

### ISSUE THREAD

**Issue by valosekj** [neuropoly/data-management#195]
After a discussion with @jcohenadad, we decided to move `GmsegChallenge` subjects outside of `sct-testing-large`.

This will include the following steps:
- [ ] modification of `sct-testing-large` dataset - removal of `GmsegChallenge` nii files and entries from `participants.tsv`
- [ ] creation of a new git-annex `gmseg-challenge` dataset
- [ ] adding all 80 subjects to the `gmseg-challenge` dataset (currently, there are only 50 `GmsegChallenge` subjects in `sct-testing-large`)
- [ ] update of the G Sheets dataset table

**Comment by valosekj** [comment]
TLDR: `sct-testing-large` contains SC and GM seg, but the provided data contains WM seg from 4 raters.

---

`sct-testing-large` currently contains 50 from the total of 80 `GmsegChallenge` subjects --> 10 UCL and 20 Montreal subjects are missing:

<details><summary>see participants.tsv</summary>

```
$ column_tab participants.tsv | grep GmsegChallenge                                                        
sub-uclGmsegChallenge001      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_01                  uclGmsegChallenge
sub-uclGmsegChallenge002      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_02                  uclGmsegChallenge
sub-uclGmsegChallenge003      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_03                  uclGmsegChallenge
sub-uclGmsegChallenge004      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_04                  uclGmsegChallenge
sub-uclGmsegChallenge005      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_05                  uclGmsegChallenge
sub-uclGmsegChallenge006      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_06                  uclGmsegChallenge
sub-uclGmsegChallenge007      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_07                  uclGmsegChallenge
sub-uclGmsegChallenge008      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_08                  uclGmsegChallenge
sub-uclGmsegChallenge009      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_09                  uclGmsegChallenge
sub-uclGmsegChallenge010      n/a  n/a  unknown           unknown        n/a            n/a           HC                                 ucl_gmsegChallenge16_10                  uclGmsegChallenge
sub-vanderbiltGmChallenge001  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_01           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge002  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_02           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge003  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_03           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge004  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_04           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge005  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_05           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge006  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_06           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge007  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_07           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge008  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_08           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge009  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_09           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge010  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_10           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge011  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_11           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge012  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_12           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge013  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_13           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge014  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_14           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge015  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_15           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge016  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_16           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge017  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_17           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge018  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_18           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge019  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_19           vanderbiltGmsegChallenge
sub-vanderbiltGmChallenge020  n/a  n/a  unknown           unknown        n/a            n/a           HC                                 vanderbilt_gmsegChallenge16_20           vanderbiltGmsegChallenge
sub-zurichGmsegChallenge001   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_01               zurichGmsegChallenge
sub-zurichGmsegChallenge002   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_02               zurichGmsegChallenge
sub-zurichGmsegChallenge003   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_03               zurichGmsegChallenge
sub-zurichGmsegChallenge004   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_04               zurichGmsegChallenge
sub-zurichGmsegChallenge005   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_05               zurichGmsegChallenge
sub-zurichGmsegChallenge006   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_06               zurichGmsegChallenge
sub-zurichGmsegChallenge007   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_07               zurichGmsegChallenge
sub-zurichGmsegChallenge008   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_08               zurichGmsegChallenge
sub-zurichGmsegChallenge009   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_09               zurichGmsegChallenge
sub-zurichGmsegChallenge010   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_10               zurichGmsegChallenge
sub-zurichGmsegChallenge011   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_11               zurichGmsegChallenge
sub-zurichGmsegChallenge012   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_12               zurichGmsegChallenge
sub-zurichGmsegChallenge013   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_13               zurichGmsegChallenge
sub-zurichGmsegChallenge014   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_14               zurichGmsegChallenge
sub-zurichGmsegChallenge015   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_15               zurichGmsegChallenge
sub-zurichGmsegChallenge016   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_16               zurichGmsegChallenge
sub-zurichGmsegChallenge017   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_17               zurichGmsegChallenge
sub-zurichGmsegChallenge018   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_18               zurichGmsegChallenge
sub-zurichGmsegChallenge019   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_19               zurichGmsegChallenge
sub-zurichGmsegChallenge020   n/a  n/a  unknown           unknown        n/a            n/a           HC                                 zurich_gmsegChallenge16_20               zurichGmsegChallenge 
```

</details>

Each `sct-testing-large/*GmsegChallenge*` subject  has T2star .nii.gz and .json file:

```
$ tree sub-uclGmsegChallenge001/anat                                                                        
sub-uclGmsegChallenge001/anat
├── sub-uclGmsegChallenge001_T2star.json
└── sub-uclGmsegChallenge001_T2star.nii.gz
```

And **_SC_** and **_GM_** seg:

```
$ tree derivatives/labels/sub-uclGmsegChallenge001/anat                                                     
derivatives/labels/sub-uclGmsegChallenge001/anat
├── sub-uclGmsegChallenge001_T2star_gmseg-manual.json
├── sub-uclGmsegChallenge001_T2star_gmseg-manual.nii.gz
├── sub-uclGmsegChallenge001_T2star_seg-manual.json
└── sub-uclGmsegChallenge001_T2star_seg-manual.nii.gz
```

However, the data I downloaded from the [provided links ](https://neuropoly.slack.com/archives/CB232HT46/p1669732366448529) contains differently structured derivatives. Namely, the **_WM_** seg from 4 raters is available for each subject:

```
$ ls -1 site1-sc01*
site1-sc01-image.nii.gz
site1-sc01-levels.txt
site1-sc01-mask-r1.nii.gz
site1-sc01-mask-r2.nii.gz
site1-sc01-mask-r3.nii.gz
site1-sc01-mask-r4.nii.gz
```

This leads me to the idea of creating a new git-annex `gmseg-challenge` dataset ([as mentioned above](https://github.com/neuropoly/data-management/issues/195#issue-1494906253)) but keeping the current `sct-testing-large/*GmsegChallenge*` subjects under `sct-testing-large` ([instead of removing them as mentioned above](https://github.com/neuropoly/data-management/issues/195#issue-1494906253)).

**Comment by jcohenadad** [comment]
Thank you for the excellent detective work @valosekj 

> However, the data I downloaded from the [provided links ](https://neuropoly.slack.com/archives/CB232HT46/p1669732366448529)contains differently structured derivatives. Namely, the WM seg from 4 raters is available for each subject:

Indeed, we modified the original naming to be BIDS compatible. Also, we created GM and SC segs out of the WM segmentations because this is what we trained our models on. 

> but keeping the current sct-testing-large/*GmsegChallenge* subjects under sct-testing-large (https://github.com/neuropoly/data-management/issues/195#issue-1494906253).

I'm not a big fan of this suggestion, because it will duplicate data, which could cause methodological issues in the future. Eg: if a student trains a model using "sct-testing-large/*GmsegChallenge" _and_ "gmseg-challenge", the same data could appear in the training and testing split. I would be inclined to discard the "sct-testing-large/*GmsegChallenge", and re-create the SC and GM labels (eg: by using flood filling algorithm to get SC, and then subtracting the SC from the WM mask).

More details about the mask generation [in the original paper from Prados et al. Neuroimage](https://www.sciencedirect.com/science/article/pii/S1053811917302185#f0025), in section "GM mask delineation". However, it would be good to contact the organizer (Ferran Prados) and ask how the GM and SC masks are generated from the WM masks, Because, e.g. if the GM mask touches the edge of the SC mask, then a flood-fill algorithm would fail.

**Comment by valosekj** [comment]
Thank you very much for the explanation! 

> I would be inclined to discard the "sct-testing-large/*GmsegChallenge", and re-create the SC and GM labels 

I agree. 
Moreover, during the first check of the data, I missed the fact that the provided segmentations actually contain both WM (pixel value `2`) and GM (pixel value `1`):

<img width="250" alt="image" src="https://user-images.githubusercontent.com/39456460/210094132-3af4bf62-da8e-4ada-bf69-75456ef0d8f4.png">

Thus, the SC and GM labels can be easily re-created using simple thresholding. --> I will do it.

**Comment by jcohenadad** [comment]
I'm in favor of updating [our conventions](https://github.com/ivadomed/ivadomed/wiki/repositories#derivatives) (also [here](https://intranet.neuro.polymtl.ca/data/dataset-curation.html#bids-template)) for derivatives/labels, see notably: https://github.com/sct-pipeline/fmri-segmentation/issues/1. 

so, in this case, it would look something like this:
~~~
derivatives/label/xxx/anat/xxx_T1w_label-SC_desc-manual_mask.nii.gz
derivatives/label/xxx/anat/xxx_T1w_label-GM_desc-manual_mask.nii.gz
derivatives/label/xxx/anat/xxx_T1w_label-WM_desc-manual_mask.nii.gz
~~~

Sources:
- https://hackmd.io/@effigies/bids-derivatives-readme#Future-Directions
- https://docs.google.com/document/d/1YG2g4UkEio4t_STIBOqYOwneLEs1emHIXbGKynx7V0Y/edit#

**Comment by valosekj** [comment]
> I'm in favor of updating [our conventions](https://github.com/ivadomed/ivadomed/wiki/repositories#derivatives) (also [here](https://intranet.neuro.polymtl.ca/data/dataset-curation.html#bids-template)) for derivatives/labels, see notably: [sct-pipeline/fmri-segmentation#1](https://github.com/sct-pipeline/fmri-segmentation/issues/1).
> 

It sounds good!

> so, in this case, it would look something like this:
> 
> ```
> derivatives/label/xxx/anat/xxx_T1w_label-SC_desc-manual_mask.nii.gz
> derivatives/label/xxx/anat/xxx_T1w_label-GM_desc-manual_mask.nii.gz
> derivatives/label/xxx/anat/xxx_T1w_label-WM_desc-manual_mask.nii.gz
> ```

I have used this convention in https://github.com/neuropoly/data-management/pull/199. Once we agree on that, I will update [ivadomed](https://github.com/ivadomed/ivadomed/wiki/repositories#derivatives) and [intranet](https://intranet.neuro.polymtl.ca/data/dataset-curation.html#bids-template) websites.

Based on "Derivative data types" section [here](https://hackmd.io/@effigies/bids-derivatives-readme#BIDS-Derivatives1), maybe we could use also `derivatives/manual_masks` instead of `derivatives/label(s)`

**Comment by jcohenadad** [comment]
> Based on "Derivative data types" section [here](https://hackmd.io/@effigies/bids-derivatives-readme#BIDS-Derivatives1), maybe we could use also `derivatives/manual_masks` instead of `derivatives/label(s)`

Excellent idea. In this case, we could get rid of the `desc-manual`:

```
derivatives/manual_labels/xxx/anat/xxx_T1w_label-SC_mask.nii.gz
derivatives/manual_labels/xxx/anat/xxx_T1w_label-GM_mask.nii.gz
derivatives/manual_labels/xxx/anat/xxx_T1w_label-WM_mask.nii.gz
```

And I suggest changing `manual_masks` for `manual_labels` in case we have disc labels (which are not masks).
